src/core/thread/mle_router.cpp

```diff
@@ -2081,46 +2081,59 @@ otError MleRouter::SendParentResponse(Child *aChild, const ChallengeTlv &aChalle
     return OT_ERROR_NONE;
 }
 
-otError MleRouter::UpdateChildAddresses(const AddressRegistrationTlv &aTlv, Child &aChild)
+otError MleRouter::UpdateChildAddresses(const Message &aMessage, uint16_t aOffset, Child &aChild)
 {
-    const AddressRegistrationEntry *entry;
+    AddressRegistrationEntry entry;
     Ip6::Address address;
     Lowpan::Context context;
-    uint8_t count;
+    Tlv tlv;
+    uint8_t registeredCount = 0;
+    uint8_t storedCount = 0;
+    uint16_t offset = aOffset;
+    uint16_t end = 0;
     char stringBuffer[Ip6::Address::kIp6AddressStringSize];
 
+    VerifyOrExit(aMessage.Read(offset, sizeof(tlv), &tlv) == sizeof(tlv));
+
+    offset += sizeof(tlv);
+    end = offset + tlv.GetLength();
```

Comment:
Validate that the TLV fits within the message (i.e. `tlv.GetLength() <= aMessage.GetLength() - aOffset`)?

Reply:
Updated. Thanks.